src/AuthorizeDotNet.php

```diff
@@ -0,0 +1,301 @@
+     * @param  \stdClass $oInvoice     The invoice object
+     * @param  string    $sSuccessUrl  The URL to go to after successful payment
+     * @param  string    $sFailUrl     The URL to go to after failed payment
+     * @param  string    $sContinueUrl The URL to go to after payment is completed
+     *
+     * @return \Nails\Invoice\Model\ChargeResponse
+     */
+    public function charge(
+        $iAmount,
+        $sCurrency,
+        $oData,
+        $oCustomData,
+        $sDescription,
+        $oPayment,
+        $oInvoice,
+        $sSuccessUrl,
+        $sFailUrl,
+        $sContinueUrl
+    ) {
+
+        $oChargeResponse = Factory::factory('ChargeResponse', 'nailsapp/module-invoice');
+
+        try {
+
+            if (!empty($oInvoice->customer->billing_email)) {
+                $sReceiptEmail = $oInvoice->customer->billing_email;
+            } else {
+                $sReceiptEmail = $oInvoice->customer->email;
+            }
+
+            if (!isset($oCustomData->payment_profile_id)) {
+                throw new \RuntimeException(
+                    'A Payment Profile ID must be supplied.'
+                );
+            }
+
+            if (!isset($oCustomData->customer_profile_id)) {
+                throw new \RuntimeException(
+                    'The supplied Payment Profile ID cannot be used without an accompanying Customer Profile ID'
+                );
+            }
+
+            $oPaymentProfile = new AuthNetAPI\PaymentProfileType();
+            $oPaymentProfile->setPaymentProfileId($oCustomData->payment_profile_id);
+
+            $oCustomerProfile = new AuthNetAPI\CustomerProfilePaymentType();
+            $oCustomerProfile->setCustomerProfileId($oCustomData->customer_profile_id);
+            $oCustomerProfile->setPaymentProfile($oPaymentProfile);
+
+            $oCharge = AuthNetAPI\TransactionRequestType();
+            $oCharge->setTransactionType('authCaptureTransaction');
+            $oCharge->setCurrencyCode($sCurrency);
+            $oCharge->setAmount($iAmount / 100);
+            $oCharge->setProfile($oCustomerProfile);
+
+            $oApiRequest = new AuthNetAPI\CreateTransactionRequest();
+            $oApiRequest->setMerchantAuthentication($this->oAuthentication);
+            $oApiRequest->setTransactionRequest($oCharge);
+
+            $oApiController = new AuthNetController\CreateTransactionController($oApiRequest);
+            $oResponse      = $oApiController->executeWithApiResponse($this->sApiMode);
+
+            if ($oResponse->getMessages()->getResultCode() === static::AUTH_NET_RESPONSE_OK) {
+
+                $oChargeResponse->setStatusComplete();
+                $oChargeResponse->setTxnId($oResponse->getTransactionResponse()->getTransId());
+
+            } else {
+
+                //  @todo: handle errors returned by the Stripe Client/API
+                $oChargeResponse->setStatusFailed(
+                    null,
```

Comment:
Remove the `@todo`. The first parameter is where we can put detailed (non user facing) error message, please add if there's something available.